app/helpers/facets_helper.rb

```diff
@@ -9,6 +9,7 @@ def get_display_facet_types
       display_facet = agg[facet_config.field]
       next if display_facet.nil? || display_facet.items.empty? || !should_render_field?(facet_config, display_facet)
       facet_type = facet_config[:facet_type] || :default
+      facet_type = facet_type.call(params) if facet_type.respond_to?(:lambda?)
```

Comment:
this checking of facet_type here occurs quite a few times throughout this code. could we instead define two distinct `subject_correlation` facet fields to distinguish between first-class and default behaviour?

Reply:
This is a good question. I'd like to discuss this further in person and figure out the best way to go ...